encode.go

```diff
@@ -61,7 +61,7 @@ func NewEncoder(w io.Writer) *Encoder {
 	})
 }
 
-// NewEncoderWithConfig is similar to NewEncoder, but allows specifying the encoder configuration.
+// NewEncoderWithConfig is similar to NewEncoder, but allows to specify the encoder configuration.
```

Comment:
I think the original is actually more gramatically correct than this change, which is not (allows who or what to specify? missing a subject)

Reply:
Something like "NewEncoderWithConfig returns a new Encoder with the given configuration" would be clearer. Also it's not clear if EncoderConfig can be null here?

Reply:
Thanks for noticing this, Kamil.

After a fresh look I see that there is even mismatch in between NewEncoder and NewDecoder documentation, so also taking your suggestions into account maybe do you think the following fixup would be good?

```diff
diff --git a/encode.go b/encode.go
index e30b18b..9955620 100644
--- a/encode.go
+++ b/encode.go
@@ -53,7 +53,9 @@ type EncoderConfig struct {
        StrictUnicode bool
 }
 
-// NewEncoder returns a new Encoder struct with default values
+// NewEncoder returns a new Encoder with default configuration.
+//
+// The encoder will emit pickle stream into w.
 func NewEncoder(w io.Writer) *Encoder {
        return NewEncoderWithConfig(w, &EncoderConfig{
                // allow both Python2 and Python3 to decode what ogórek produces by default
@@ -61,7 +63,9 @@ func NewEncoder(w io.Writer) *Encoder {
        })
 }
 
-// NewEncoderWithConfig is similar to NewEncoder, but allows to specify the encoder configuration.
+// NewEncoderWithConfig is similar to NewEncoder, but returns the encoder with specified configuration.
+//
+// config must be !nil.
 func NewEncoderWithConfig(w io.Writer, config *EncoderConfig) *Encoder {
        return &Encoder{w: w, config: config}
 }
diff --git a/ogorek.go b/ogorek.go
index 46ce084..f7b9096 100644
--- a/ogorek.go
+++ b/ogorek.go
@@ -191,12 +191,16 @@ type DecoderConfig struct {
        StrictUnicode bool
 }
 
-// NewDecoder constructs a new Decoder which will decode the pickle stream in r.
+// NewDecoder returns a new Decoder with default configuration.
+//
+// The decoder will decode the pickle stream in r.
 func NewDecoder(r io.Reader) *Decoder {
        return NewDecoderWithConfig(r, &DecoderConfig{})
 }
 
-// NewDecoderWithConfig is similar to NewDecoder, but allows specifying decoder configuration.
+// NewDecoderWithConfig is similar to NewDecoder, but returns decoder with specified configuration.
+//
+// config must be !nil.
 func NewDecoderWithConfig(r io.Reader, config *DecoderConfig) *Decoder {
        reader := bufio.NewReader(r)
        return &Decoder{

```

Reply:
I like it, it just needs to say "_the_ default configuration" and "_the_ specified configuration". 

Any comment on what happens if the given configuration is nil or if that is even valid?

Reply:
Thanks. Alright, I've added "the" and explicit "not nil" instead of "!nil" as you suggest and pushed the amended patch. Hope it is ok now.